openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -110,7 +112,8 @@ def update_user_preferences(requesting_user, update, username=None):
         PreferenceUpdateError: the operation failed when performing the update.
         UserAPIInternalError: the operation failed due to an unexpected error.
     """
-    existing_user = _get_user(requesting_user, username)
+    if not existing_user:
```

Comment:
Looking at this more, I do have some angst over the fact that all the other API methods take username instead of user. It would be nice for them to be consistent, even though update_user_preferences isn't widely used. 

However, I find the fourth optional argument a very confusing addition to the API, especially since it is not at all clear what will happen if you supply BOTH username and existing_user. The new documentation isn't correct, in that requesting_user.username is only used if username is not supplied (assuming _get_user is called).

Given all of that, I guess I am still thinking it is better to require a user object to be passed in instead of a username (unless we want to do type-checking of the last variable and allow it to be either a string or a user object... I'm not sure how acceptable that is in Python). If there is a strong reason to have a variant of the method later with username instead of user, we could create it at that time.